### PR TITLE
Adding Snap package

### PR DESCRIPTION
**Pull request by joedborg**
This will allow Linux users, with snap package manager installed, to install helm with
`snap install helm`

### CODE REVIEW

**Comment by joedborg** [comment]
LFCLA signed

**Comment by technosophos** [comment]
Ooo... this is an interesting one! We don't currently have any policies about whether we can put config for external package managers in the repo. Does anyone object to this?

**Comment by bacongobbler** [comment]
I'm all for this as long as @joedborg is the maintainer for the helm snap package. :)

**Comment by joedborg** [comment]
Hey guys, thanks for the positive notes on this.

In terms of maintainence, it should be a change of "volume" and "source".  You can then register this repository at https://build.snapcraft.io and new releases will auto build.

I can take care of this if necessary but would be nice to see it taken over if possible.

**Comment by technosophos** [comment]
Do we need to add some documentation into the `docs/install.md` to explain this?

**Comment by thomastaylor312** [comment]
@joedborg I think we are ok with adding this in so long as we have some documentation in `docs/install.md` to describe the new functionality. Is there any way you could add that in?

**Comment by jascott1** [comment]
This appears to be located at `snap/`. Should we create a  `community/packages` or similar to hold artifacts like these?

**Comment by thomastaylor312** [comment]
@jascott1 I think @adamreese said he was going to go around and explain what we had decided to do with package management stuff in our developer meeting

**Comment by joedborg** [comment]
hey @thomastaylor312, yes I can add that in.  Just FYI, the "upstream" version is here (for now) https://github.com/snapcrafters/helm

**Comment by bacongobbler** [comment]
So as far as I remember what we discussed in the developer call, we were not going to move forward with hosting "package-manager-specific" code in Helm core. We cannot keep up with the maintenance of an endlessly sprawling number of ways to install Helm. However, third party package managers are more than welcome to create repositories at [kubernetes-helm](https://github.com/kubernetes-helm) as part of the community shift.

I still think @adamreese had more opinions on this topic, but from memory I recall the story moving forward would be:

a) The community will be responsible for supporting the infrastructure of their own package manager
b) Helm Core Maintainers can create repositories at https://github.com/kubernetes-helm for hosting these package manager-specific code repositories (upon request/approval)
c) We will continue to release Helm as necessary, circling in package manager maintainers into upcoming Helm releases through the roadmap, emails, weekly Helm dev calls etc.

The TL;DR is that it's up to the community to maintain these third party packages. We will continue to distribute releases through our usual release process, but alternative installation methods are up to the community to support.

**Comment by joedborg** [comment]
Hey @bacongobbler thanks for the feedback!  Just a couple of questions based on them.

1) I think the community is more than happy to maintain these (as has been done in this case).  However, I believe it's l better from upstream for 2 reasons. First is that all the docs etc point to the right place.  Second is that cadence of release is matched with the package manager.  I do completely understand that this is more work for you guys though.

2) I've just pushed the latest helm version to snap, so if anyone fancies trying to break it, please feel free to `sudo snap install helm`

**Comment by bacongobbler** [comment]
> I do completely understand that this is more work for you guys though.

At the end of the day, this is the core reason why we're deciding against merging this. We don't have the time to maintain all the things we don't use.

You are more than welcome to host it at the kubernetes-helm org and get some snap users to support the snap package for Helm. Feel free to reach out during one the weekly dev calls if you want to get help with setting that repo up, and perhaps see if you can round up some users in the #kubernetes-helm channel [in Slack](https://kubernetes.slack.com) to take over maintenance.

Thanks!
